### DISCUSSION TITLE
What does this finalPath filter do?

### DISCUSSION THREAD

**Discussion by notflip** [directus-labs/agency-os#35] [Q&A] [answered]
I'm wondering what this filter does, and why it is needed, **what problem does this solve?**

https://github.com/directus-community/agency-os/blob/26c39ec13caa458e6b1bb6dc52cece4a005ff876/pages/%5B...permalink%5D.vue#L8

**Comment by bryantgillespie** [comment, marked as answer]
It's super common to have parent pages and subpages like...
- https://mysite.com/about
- https://mysite.com/about/values

To achieve that without using recursive relationships on the `pages` collection within Directus, one way to solve it is by instead using a permalink, instead of a slug.

![ScreenShot 2023-10-25 at 15 43 44@2x](https://github.com/directus-community/agency-os/assets/23302570/4c9cfb74-214b-4972-a6c9-52a47461975e)

Within Nuxt, the `...` within [...permalink] makes it all [catch all route](https://nuxt.com/docs/guide/directory-structure/pages#catch-all-route) that will catch both urls like above.

So the filter ensures the the route.path within Nuxt returns the homepage route, urls with any trailing slashes, and all others match the structure used within the pages collection inside Directus.

**Reply by notflip** [reply]
That's a good point! Thanks @bryantgillespie, I'm going to swap the slugs out in my running projects, permalink makes more sense

**Reply by bryantgillespie** [reply]
For more general collections like `pages` I think it makes a lot of sense 👍 .
